Kitodo-API/src/main/java/org/kitodo/api/ugh/DocStructInterface.java

```diff
@@ -0,0 +1,628 @@
+     * @param identifierField
+     *            meta data field that holds the identifier to locate the child
+     * @param identifier
+     *            identifier of the child to locate
+     * @return the child, if found
+     * @throws NoSuchElementException
+     *             if no matching child is found
+     */
+    DocStructInterface getChild(String type, String identifierField, String identifier);
+
+    /**
+     * Returns all meta-data types that shall be displayed even if they have no
+     * value.
+     *
+     * <p>
+     * Comprises all meta-data types whose attribute
+     * {@code defaultDisplay="true"} is set in the {@code Preferences}. Hidden
+     * meta-data, whose {@code MetadataType} starts with the
+     * {@code HIDDEN_METADATA_CHAR}, will not be included.
+     *
+     * @return all meta-data group types that shall always be displayed
+     */
+    List<MetadataTypeInterface> getDisplayMetadataTypes();
+
+    /**
+     * Returns the image name.
+     *
+     * @return the image name
+     */
+    String getImageName();
+
+    /**
+     * Returns the next child in the list of all children. If the given
+     * {@code DocStruct} isn’t a child of the current instance, {@code null} is
+     * returned.
+     *
+     * @param inChild
+     *            {@code DocStruct} whose successor shall be returned
+     * @return the next {@code DocStruct} after {@code inChild}, {@code null}
+     *         otherwise
+     */
+    DocStructInterface getNextChild(DocStructInterface tempDS);
+
+    /**
+     * Returns the parent of this instance. Returns {@code null} if this
+     * instance is the root of the tree.
+     *
+     * @return the parent, if any
+     */
+    DocStructInterface getParent();
+
+    /**
+     * Returns all meta-data types that can be added to this instance. Includes
+     * meta-data groups, whose {@code MetadataGroupType} starts with the
+     * {@code HIDDEN_METADATA_CHAR}.
+     *
+     * <p>
+     * This method considers already added {@code Metadata}, so meta-data types
+     * which can only be available once cannot be added a second time. Therefore
+     * these {@code MetadataType}s will not be included in this list.
+     *
+     * @return all meta-data types that can be added to this instance. A return
+     *         type of {@code Collection<>} would be sufficient.
+     */
+    List<MetadataTypeInterface> getPossibleMetadataTypes();
+
+    /**
+     * Get the type of this DocStruct.
+     *
+     * @return the type of this DocStruct
+     */
+    DocStructTypeInterface getDocStructType();
+
+    /**
+     * Returns whether a {@code DocStruct} of the given {@code DocStructType} is
+     * allowed to be added to this instance.
+     *
+     * @param inType
+     *            the {@code DocStructType} in question
+     * @return true, if {@code DocStruct} of this type can be added; otherwise
+     *         false
+     */
+    boolean isDocStructTypeAllowedAsChild(DocStructTypeInterface type);
```

Comment:
Please adjust the JavaDoc or method signature as they are different.

Reply:
I don’t see what is wrong here?

Reply:
In JavaDoc method parameter is called `inType` and in method the parameter is called `type`.

Reply:
Github is showing an outdated commit here. This is already fixed.

Reply:
I see after I downloaded the patch file for this pull request.